core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandTest.scala

```diff
@@ -27,54 +27,106 @@ import kafka.utils.CoreUtils._
 import kafka.utils.TestUtils._
 import kafka.utils.{CoreUtils, Logging, TestUtils}
 import kafka.zk.{AdminZkClient, KafkaZkClient, ZooKeeperTestHarness}
+import org.apache.kafka.clients.CommonClientConfigs
+import org.apache.kafka.clients.admin.{AdminClient, DescribeReplicaLogDirsResult}
+import org.apache.kafka.common.network.ListenerName
+import org.apache.kafka.common.security.auth.SecurityProtocol
 import org.easymock.EasyMock._
 import org.easymock.{Capture, CaptureType, EasyMock}
 import org.junit.{After, Before, Test}
 import org.junit.Assert.{assertEquals, assertFalse, assertNull, assertTrue}
 import org.scalatest.Assertions.fail
 
 import scala.collection.JavaConverters._
-import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.{TopicPartition, TopicPartitionReplica}
+import org.apache.zookeeper.KeeperException.UnimplementedException
 
 import scala.collection.mutable
 import scala.collection.Seq
+import scala.util.Try
 
+// Even though we test with a mock ServiceClient where possible, use the ZooKeeperTestHarness
+// for use by the TestUtils below.
 class ReassignPartitionsCommandTest extends ZooKeeperTestHarness with Logging {
   var servers: Seq[KafkaServer] = Seq()
   var calls = 0
 
+  @Before
+  def setup(): Unit = {
+    calls = 0
+  }
+
   @After
   override def tearDown(): Unit = {
     TestUtils.shutdownServers(servers)
     super.tearDown()
   }
 
+  // Base Test Service Client with minimal functionality, just enough for our tests to
+  // override
+  class TestServiceClientBase extends ReassignCommandService {
+    override def getBrokerIdsInCluster: Seq[Int] = throw new UnimplementedException()
```

Comment:
Should we use our own exception? I don't think we should depend on a ZK library exception